From eab7a74dab02a83111fcdf1860ca796a38e2a198 Mon Sep 17 00:00:00 2001
From: Tony Cheng <tony.cheng@amd.com>
Date: Sat, 17 Sep 2016 00:02:59 -0400
Subject: [PATCH 1099/1722] drm/amd/dal: remove deadcode start_gtc_sync
 stop_gtc_sync

- GTC never productize due to lack of Rx support.  this feature isn't implement in DAL3 anyways.  remove

Change-Id: I536392862450aa68fef48d6ec44a8e3063ef7ee5
Signed-off-by: Tony Cheng <tony.cheng@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 .../gpu/drm/amd/dal/dc/adapter/adapter_service.c   |  1 -
 .../drm/amd/dal/dc/dce110/dce110_hw_sequencer.c    |  2 -
 .../gpu/drm/amd/dal/dc/gpu/dc_clock_generator.c    | 21 ---------
 .../gpu/drm/amd/dal/dc/gpu/dc_clock_generator.h    |  9 ----
 drivers/gpu/drm/amd/dal/dc/i2caux/aux_engine.h     |  4 --
 .../amd/dal/dc/i2caux/dce110/aux_engine_dce110.c   | 20 --------
 .../amd/dal/dc/i2caux/dce110/aux_engine_dce110.h   |  3 --
 .../drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.c | 19 --------
 .../drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.h |  3 --
 drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c         | 54 ----------------------
 .../amd/dal/include/adapter_service_interface.h    |  1 -
 .../amd/dal/include/dc_clock_generator_interface.h |  9 ----
 drivers/gpu/drm/amd/dal/include/i2caux_interface.h |  8 ----
 drivers/gpu/drm/amd/dal/include/set_mode_types.h   | 12 -----
 14 files changed, 166 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c b/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
index 9883920..dc0c99b 100644
--- a/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
+++ b/drivers/gpu/drm/amd/dal/dc/adapter/adapter_service.c
@@ -117,7 +117,6 @@ static const struct feature_source_entry feature_entry_table[] = {
 	{FEATURE_ALLOW_EDP_RESOURCE_SHARING, false, true},
 	{FEATURE_SUPPORT_DP_YUV, false, true},
 	{FEATURE_SUPPORT_DP_Y_ONLY, false, true},
-	{FEATURE_DISABLE_DP_GTC_SYNC, true, true},
 	{FEATURE_MODIFY_TIMINGS_FOR_WIRELESS, false, true},
 	{FEATURE_DCP_BIT_DEPTH_REDUCTION_MODE, 0, false},
 	{FEATURE_DCP_DITHER_MODE, 0, false},
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
index f294253..106cc99 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
@@ -1108,8 +1108,6 @@ static void set_display_clock(struct validate_context *context)
 	/* TODO: When changing display engine clock, DMCU WaitLoop must be
 	 * reconfigured in order to maintain the same delays within DMCU
 	 * programming sequences. */
-
-	/* TODO: Start GTC counter */
 }
 
 static uint32_t compute_pstate_blackout_duration(
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.c b/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.c
index b3b0f99..67cdc8f 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.c
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.c
@@ -53,27 +53,6 @@ bool dal_dc_clock_generator_get_dp_ref_clk_ds_params(
 	return dc_clk_gen->funcs->get_dp_ref_clk_ds_params(dc_clk_gen, params);
 }
 
-bool dal_dc_clock_generator_enable_gtc_counter(
-	struct dc_clock_generator *dc_clk_gen,
-	uint32_t dprefclk)
-{
-	return dc_clk_gen->funcs->enable_gtc_counter(dc_clk_gen, dprefclk);
-}
-
-void dal_dc_clock_generator_disable_gtc_counter(
-	struct dc_clock_generator *dc_clk_gen)
-{
-	dc_clk_gen->funcs->disable_gtc_counter(dc_clk_gen);
-}
-
-void dal_dc_clock_generator_set_gtc_group_offset(
-	struct dc_clock_generator *dc_clk_gen,
-	enum gtc_group group_num,
-	uint32_t offset)
-{
-	dc_clk_gen->funcs->set_gtc_group_offset(dc_clk_gen, group_num, offset);
-}
-
 void dal_dc_clock_generator_base_set_display_pipe_mapping(
 	struct dc_clock_generator *base,
 	struct dccg_mapping_params *params)
diff --git a/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.h b/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.h
index d1bf1af..bc0419c 100644
--- a/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.h
+++ b/drivers/gpu/drm/amd/dal/dc/gpu/dc_clock_generator.h
@@ -38,15 +38,6 @@ struct dc_clock_generator_funcs {
 	bool (*get_dp_ref_clk_ds_params)(
 		struct dc_clock_generator *dc_clk_gen,
 		struct dccg_dp_ref_clk_ds_params *params);
-	bool (*enable_gtc_counter)(
-		struct dc_clock_generator *dc_clk_gen,
-		uint32_t dprefclk);
-	void (*disable_gtc_counter)(
-		struct dc_clock_generator *dc_clk_gen);
-	void (*set_gtc_group_offset)(
-		struct dc_clock_generator *dc_clk_gen,
-		enum gtc_group group_num,
-		uint32_t offset);
 };
 struct dc_clock_generator {
 	const struct dc_clock_generator_funcs *funcs;
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/aux_engine.h b/drivers/gpu/drm/amd/dal/dc/i2caux/aux_engine.h
index f8ec4ba..7c45b6b 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/aux_engine.h
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/aux_engine.h
@@ -77,10 +77,6 @@ struct aux_engine_funcs {
 	void (*configure)(
 		struct aux_engine *engine,
 		union aux_config cfg);
-	bool (*start_gtc_sync)(
-		struct aux_engine *engine);
-	void (*stop_gtc_sync)(
-		struct aux_engine *engine);
 	void (*submit_channel_request)(
 		struct aux_engine *engine,
 		struct aux_request_transaction_data *request);
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
index cb43f83..f3f3f13 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.c
@@ -220,19 +220,6 @@ static void configure(
 	dm_write_reg(engine->base.ctx, addr, value);
 }
 
-static bool start_gtc_sync(
-	struct aux_engine *engine)
-{
-	/*TODO*/
-	return false;
-}
-
-static void stop_gtc_sync(
-	struct aux_engine *engine)
-{
-	/*TODO*/
-}
-
 #define COMPOSE_AUX_SW_DATA_16_20(command, address) \
 	((command) | ((0xF0000 & (address)) >> 16))
 
@@ -701,9 +688,6 @@ static const int32_t aux_channel_offset[] = {
 static const struct aux_engine_funcs aux_engine_funcs = {
 	.destroy = destroy,
 	.acquire_engine = acquire_engine,
-	.configure = configure,
-	.start_gtc_sync = start_gtc_sync,
-	.stop_gtc_sync = stop_gtc_sync,
 	.submit_channel_request = submit_channel_request,
 	.process_channel_reply = process_channel_reply,
 	.get_channel_status = get_channel_status,
@@ -742,10 +726,6 @@ static bool construct(
 	engine->addr.aux_sw_control = mmAUX_SW_CONTROL + offset;
 	engine->addr.aux_interrupt_control = mmAUX_INTERRUPT_CONTROL + offset;
 	engine->addr.aux_sw_status = mmAUX_SW_STATUS + offset;
-	engine->addr.aux_gtc_sync_control = mmAUX_GTC_SYNC_CONTROL + offset;
-	engine->addr.aux_gtc_sync_status = mmAUX_GTC_SYNC_STATUS + offset;
-	engine->addr.aux_gtc_sync_controller_status =
-		mmAUX_GTC_SYNC_CONTROLLER_STATUS + offset;
 
 	engine->timeout_period = aux_init_data->timeout_period;
 
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.h b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.h
index ec6899e..d2dd4d9 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.h
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce110/aux_engine_dce110.h
@@ -37,9 +37,6 @@ struct aux_engine_dce110 {
 		uint32_t aux_sw_control;
 		uint32_t aux_interrupt_control;
 		uint32_t aux_sw_status;
-		uint32_t aux_gtc_sync_control;
-		uint32_t aux_gtc_sync_status;
-		uint32_t aux_gtc_sync_controller_status;
 	} addr;
 	uint32_t timeout_period;
 };
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.c b/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.c
index 34a37e1..b297ac0 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.c
@@ -173,19 +173,6 @@ static void configure(
 	dm_write_reg(engine->base.ctx, addr, value);
 }
 
-static bool start_gtc_sync(
-	struct aux_engine *engine)
-{
-	/* TODO */
-	return false;
-}
-
-static void stop_gtc_sync(
-	struct aux_engine *engine)
-{
-	/* TODO */
-}
-
 #define COMPOSE_AUX_SW_DATA_16_20(command, address) \
 	((command) | ((0xF0000 & (address)) >> 16))
 
@@ -655,8 +642,6 @@ static const struct aux_engine_funcs aux_engine_funcs = {
 	.destroy = destroy,
 	.acquire_engine = acquire_engine,
 	.configure = configure,
-	.start_gtc_sync = start_gtc_sync,
-	.stop_gtc_sync = stop_gtc_sync,
 	.submit_channel_request = submit_channel_request,
 	.process_channel_reply = process_channel_reply,
 	.get_channel_status = get_channel_status,
@@ -693,10 +678,6 @@ static bool construct(
 	engine->addr.AUX_SW_CONTROL = mmAUX_SW_CONTROL + offset;
 	engine->addr.AUX_INTERRUPT_CONTROL = mmAUX_INTERRUPT_CONTROL + offset;
 	engine->addr.AUX_SW_STATUS = mmAUX_SW_STATUS + offset;
-	engine->addr.AUX_GTC_SYNC_CONTROL = mmAUX_GTC_SYNC_CONTROL + offset;
-	engine->addr.AUX_GTC_SYNC_STATUS = mmAUX_GTC_SYNC_STATUS + offset;
-	engine->addr.AUX_GTC_SYNC_CONTROLLER_STATUS =
-		mmAUX_GTC_SYNC_CONTROLLER_STATUS + offset;
 
 	engine->timeout_period = arg->timeout_period;
 
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.h b/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.h
index 8523c45..3319cad 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.h
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/dce80/aux_engine_dce80.h
@@ -35,9 +35,6 @@ struct aux_engine_dce80 {
 		uint32_t AUX_SW_CONTROL;
 		uint32_t AUX_INTERRUPT_CONTROL;
 		uint32_t AUX_SW_STATUS;
-		uint32_t AUX_GTC_SYNC_CONTROL;
-		uint32_t AUX_GTC_SYNC_STATUS;
-		uint32_t AUX_GTC_SYNC_CONTROLLER_STATUS;
 	} addr;
 	uint32_t timeout_period;
 };
diff --git a/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c b/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c
index 73db3d4..b80dfa4 100644
--- a/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c
+++ b/drivers/gpu/drm/amd/dal/dc/i2caux/i2caux.c
@@ -289,60 +289,6 @@ static bool get_hw_supported_ddc_line(
 	return true;
 }
 
-bool dal_i2caux_start_gtc_sync(
-	struct i2caux *i2caux,
-	struct ddc *ddc)
-{
-	enum gpio_ddc_line line;
-
-	struct aux_engine *engine;
-
-	bool result;
-
-	if (!get_hw_supported_ddc_line(ddc, &line))
-		return false;
-
-	engine = i2caux->aux_engines[line];
-
-	if (!engine)
-		return false;
-
-	if (!engine->base.funcs->acquire(&engine->base, ddc))
-		return false;
-
-	result = engine->funcs->start_gtc_sync(engine);
-
-	i2caux->funcs->release_engine(i2caux, &engine->base);
-
-	return result;
-}
-
-bool dal_i2caux_stop_gtc_sync(
-	struct i2caux *i2caux,
-	struct ddc *ddc)
-{
-	enum gpio_ddc_line line;
-
-	struct aux_engine *engine;
-
-	if (!get_hw_supported_ddc_line(ddc, &line))
-		return false;
-
-	engine = i2caux->aux_engines[line];
-
-	if (!engine)
-		return false;
-
-	if (!engine->base.funcs->acquire(&engine->base, ddc))
-		return false;
-
-	engine->funcs->stop_gtc_sync(engine);
-
-	i2caux->funcs->release_engine(i2caux, &engine->base);
-
-	return true;
-}
-
 void dal_i2caux_configure_aux(
 	struct i2caux *i2caux,
 	struct ddc *ddc,
diff --git a/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h b/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h
index 5b9d364..088529d 100644
--- a/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h
+++ b/drivers/gpu/drm/amd/dal/include/adapter_service_interface.h
@@ -81,7 +81,6 @@ enum adapter_feature_id {
 	FEATURE_ALLOW_EDP_RESOURCE_SHARING,
 	FEATURE_SUPPORT_DP_YUV,
 	FEATURE_SUPPORT_DP_Y_ONLY,
-	FEATURE_DISABLE_DP_GTC_SYNC,
 	FEATURE_NO_HPD_LOW_POLLING_VCC_OFF, /* 20th */
 	FEATURE_ENABLE_DFS_BYPASS,
 	FEATURE_LB_HIGH_RESOLUTION,
diff --git a/drivers/gpu/drm/amd/dal/include/dc_clock_generator_interface.h b/drivers/gpu/drm/amd/dal/include/dc_clock_generator_interface.h
index b7fb9ff..a5fbaf0 100644
--- a/drivers/gpu/drm/amd/dal/include/dc_clock_generator_interface.h
+++ b/drivers/gpu/drm/amd/dal/include/dc_clock_generator_interface.h
@@ -64,14 +64,5 @@ void dal_dc_clock_generator_set_display_pipe_mapping(
 bool dal_dc_clock_generator_get_dp_ref_clk_ds_params(
 	struct dc_clock_generator *dc_clk_gen,
 	struct dccg_dp_ref_clk_ds_params *params);
-bool dal_dc_clock_generator_enable_gtc_counter(
-	struct dc_clock_generator *dc_clk_gen,
-	uint32_t dprefclk);
-void dal_dc_clock_generator_disable_gtc_counter(
-	struct dc_clock_generator *dc_clk_gen);
-void dal_dc_clock_generator_set_gtc_group_offset(
-	struct dc_clock_generator *dc_clk_gen,
-	enum gtc_group group_num,
-	uint32_t offset);
 
 #endif /* __DC_CLOCK_GENERATOR_INTERFACE_H__ */
diff --git a/drivers/gpu/drm/amd/dal/include/i2caux_interface.h b/drivers/gpu/drm/amd/dal/include/i2caux_interface.h
index 6e44f19..b23e6c0 100644
--- a/drivers/gpu/drm/amd/dal/include/i2caux_interface.h
+++ b/drivers/gpu/drm/amd/dal/include/i2caux_interface.h
@@ -79,14 +79,6 @@ bool dal_i2caux_submit_aux_command(
 	struct ddc *ddc,
 	struct aux_command *cmd);
 
-bool dal_i2caux_start_gtc_sync(
-	struct i2caux *i2caux,
-	struct ddc *ddc);
-
-bool dal_i2caux_stop_gtc_sync(
-	struct i2caux *i2caux,
-	struct ddc *ddc);
-
 void dal_i2caux_configure_aux(
 	struct i2caux *i2caux,
 	struct ddc *ddc,
diff --git a/drivers/gpu/drm/amd/dal/include/set_mode_types.h b/drivers/gpu/drm/amd/dal/include/set_mode_types.h
index 93aa534..d18210f 100644
--- a/drivers/gpu/drm/amd/dal/include/set_mode_types.h
+++ b/drivers/gpu/drm/amd/dal/include/set_mode_types.h
@@ -28,18 +28,6 @@
 
 #include "dc_types.h"
 
-/* GTC group number */
-enum gtc_group {
-	GTC_GROUP_DISABLED,
-	GTC_GROUP_1,
-	GTC_GROUP_2,
-	GTC_GROUP_3,
-	GTC_GROUP_4,
-	GTC_GROUP_5,
-	GTC_GROUP_6,
-	GTC_GROUP_MAX
-};
-
 /* Info frame packet status */
 enum info_frame_flag {
 	INFO_PACKET_PACKET_INVALID = 0,
-- 
2.7.4

